dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat

```diff
@@ -520,9 +520,24 @@ goto:eof
 
   @REM Command-line parsing of -Xshareclasses does not allow "," in cacheDir.
   if "!WLP_OUTPUT_DIR:,=!" == "!WLP_OUTPUT_DIR!" (
+
     @REM Skip if Xshareclasses is defined in IBM_JAVA_OPTIONS/OPENJ9_JAVA_OPTIONS
+    @REM First check if SPECIFIED_JAVA_OPTIONS is undefined as the second test does not seem to work with undefined variables
+    if NOT defined SPECIFIED_JAVA_OPTIONS (
+      set SHARE_CLASSES=true
+    )
     if "!SPECIFIED_JAVA_OPTIONS:Xshareclasses=!" == "!SPECIFIED_JAVA_OPTIONS!" (
-      set SERVER_IBM_JAVA_OPTIONS=-Xshareclasses:name=liberty-%%u,nonfatal,cacheDir="%WLP_OUTPUT_DIR%\.classCache" -XX:ShareClassesEnableBCI -Xscmx80m !SPECIFIED_JAVA_OPTIONS!
+      set SHARE_CLASSES=true
+    )
+ 
+    if "!SHARE_CLASSES!" == "true" (
+      @REM Set -Xscmx
+      if "debug" == "%ACTION%" (
+        set XSCMX_VAL="130m"
+      ) else (
+        set XSCMX_VAL="80m"
+      )
+      set SERVER_IBM_JAVA_OPTIONS=-Xshareclasses:name=liberty-%%u,nonfatal,cacheDir="%WLP_OUTPUT_DIR%\.classCache" -XX:ShareClassesEnableBCI -Xscmx!XSCMX_VAL! !SPECIFIED_JAVA_OPTIONS!
```

Comment:
Similarly you should be able to use `%` here instead of `!`

Reply:
Changing to % is not working for me.